node-deb

```diff
@@ -683,25 +683,39 @@ find "$@" -type f -print0 | {
   done
 }
 
-if [ -d "$source_dir/node_modules" ]; then
-  npm ls --parseable --prod | sed -e "s/$(escape "$source_dir")//g" | grep -Ev '^$' | sed -e 's:/node_modules/::g' | {
-    while IFS="\n" read -r -d '' dir; do
-      log_debug "Copying dir: $dir"
-      cp -rf "$source_dir/node_modules/$dir" "$deb_dir/usr/share/$package_name/app/node_modules/"
-    done
-  }
-fi
+app_dir="$deb_dir/usr/share/$package_name/app"
 
-if ! [ -f "$deb_dir/usr/share/$package_name/app/package.json" ]; then
+if ! [ -f "$app_dir/package.json" ]; then
   log_info "Including 'package.json' in the Debian package."
   cp -pf './package.json' "$deb_dir/usr/share/$package_name/app/"
 fi
 
-if [ -f './npm-shrinkwrap.json' ] && ! [ -f "$deb_dir/usr/share/$package_name/app/npm-shrinkwrap.json" ]; then
+if [ -f './npm-shrinkwrap.json' ] && ! [ -f "$app_dir/npm-shrinkwrap.json" ]; then
   log_info "Including 'npm-shrinkwrap.json' in the Debian package."
-  cp -pf './npm-shrinkwrap.json' "$deb_dir/usr/share/$package_name/app/"
+  cp -pf './npm-shrinkwrap.json' "$app_dir"
+fi
+
+if [ -f './yarn.lock' ] && ! [ -f "$app_dir/yarn.lock" ]; then
+  log_info "Including 'yarn.lock' in the Debian package."
+  cp -pf './yarn.lock' "$app_dir"
 fi
 
+current_dir="$(pwd)"
+cd "$app_dir"
+if [ -f "./yarn.lock" ]; then
+  log_info "Building node_modules dir with yarn"
+  yarn install --no-progress --production 2>&1 > /dev/null | sed "s/^/YARN: /"
+else
+  log_info "Building node_modules dir with npm"
+  npm install --no-progress --production  2>&1 > /dev/null | sed "s/^/NPM:  /"
```

Comment:
Ah yes, I see the `cd` there. So if we were going to keep this, we should wrap it in a function so we use a subshell to hop down into the dir and pop back up when it's done. I'm going to look at the other PR and compare the two on how to best handle this.